pycon.md

```diff
@@ -228,41 +242,50 @@ This will result in installing incompatible versions.
 
 # Defensive packaging
 
-Handle everything that can happen, then deal with everything that can't.
+Handle everything that can happen,  
+deal with everything that can't.
 
 -- 
 
-## Good error messages
+## Principle #0
+
+Update your documentation and scripts.
 
-Keep setup.py python 2 compatible. 
+## Principle #1
 
-Fails **before** setup()
+Keep all package entrances python 2 compatible,  
+    but have them err early.
 
-If you hit this case the most probable reason : ** pip is out of date**. 
+## Principle #2 
+
+For clear error messages,  
+    use multiple lines.
 
 --
 
-**Ask user to update pip, ** Do not let installation finish.
+### Use `$ pip install .`
+
+Do not use `setup.py <...>` directly
+
+Update your documentation and scripts to use `pip install [-e] .` 
+
+**NB**:Invoking `setup.py` directly will *not* respect the `requires_python`.
 
 -- 
 
-```python
-if sys.version_info < (3, 3):
-    error = """
-IPython 6.0+ does not support Python 2.6, 2.7, 3.0,
-3.1, or 3.2. Beginning with IPython 6.0, Python 3.3
-and above is required.
+### Keep ``setup.py`` python 2 compatible. 
 
-This may be due to an out of date pip.
+If installation fails **before** `setup()`, the most probable reason: 
```

Comment:
No this is incorrect. If `setup.py` is executes on Python 2, then it means that most likely Pip is out of date.